test/SerialIOTest.cpp

```diff
@@ -30,6 +30,10 @@
 #include <tuple>
 #include <vector>
 
+#if __GNUC__ >= 11
+#include <filesystem>
```

Comment:
Can we use an auxiliary function for this?
Filesystem is tricky and sometimes needs also extra link flags like `-lstdc++fs`, that I don't yet want to get into.

Reply:
Note that the `__GNUC__` macro **does not mean GCC** specifically and is also set by clang, icpc and others.

Reply:
Ah, good catch about the `__GNUC__` thing, will fix

I can have a look if I can do the same test with `fstat` or similar as well, might also go with this. 
I only need this test to ensure that the right API call is chosen by the ADIOS2 backend, so restricting it to a single compiler set that implements `<filesystem>` well should be ok as an alternative